public/app/features/alerting/unified/state/AlertmanagerContext.tsx

```diff
@@ -33,9 +36,46 @@ const AlertmanagerProvider = ({ children, accessType, alertmanagerSourceName }:
   const queryParams = locationService.getSearch();
   const updateQueryParams = locationService.partial;
   const allAvailableAlertManagers = useAlertManagersByPermission(accessType);
-  const availableAlertManagers = allAvailableAlertManagers.availableInternalDataSources.concat(
-    allAvailableAlertManagers.availableExternalDataSources
-  );
+  const isExtraConfigEnabled = config.featureToggles.alertingImportAlertmanagerUI ?? false;
+
+  const { data: grafanaConfig } = useAlertmanagerConfig(isExtraConfigEnabled ? GRAFANA_RULES_SOURCE_NAME : undefined, {
```

Comment:
Shouldn't we use `EXTRA_CONFIG_UID` instead of `GRAFANA_RULES_SOURCE_NAME`?

Reply:
To check if any extra config exists, we fetch the standard Grafana Alertmanager configuration to get the extra_config array, that's the `GRAFANA_RULES_SOURCE_NAME` is used here.